src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -3280,20 +3241,7 @@ bool AdaptDispatch::HardReset()
         _macroBuffer = nullptr;
     }
 
-    // If we're in a conpty, we need flush this RIS sequence to the connected
-    // terminal application, but we also need to follow that up with a DECSET
-    // sequence to re-enable the modes that we require (namely win32 input mode
-    // and focus event mode). It's important that this is kept in sync with the
-    // VtEngine::RequestWin32Input method which requests the modes on startup.
-    if (_api.IsConsolePty())
-    {
-        auto& stateMachine = _api.GetStateMachine();
-        if (stateMachine.FlushToTerminal())
-        {
-            auto& engine = stateMachine.Engine();
-            engine.ActionPassThroughString(L"\033[?9001h\033[?1004h");
-        }
-    }
+    _api.GetStateMachine().InjectSequence(InjectionType::RIS);
```

Comment:
📝 Does this "RIS" injection include Win32IM and Focus Mode?

Reply:
I think that the VT parser shouldn't have to be aware about ConPTY internals, as otherwise it wouldn't strictly be a parser anymore. That's why it only exposes what it encountered without specifying what action needs to be taken.

The translation to an action is then handled in `_stream.cpp`.